### DISCUSSION TITLE
Migrating auth from Firebase to Supabase

### DISCUSSION THREAD

**Discussion by marcelscruz** [supabase/supabase#1907] [Questions] [answered]
I'm about to start migrating Firebase auth and Firestore to Supabase.

Migrating Firestore doesn't seem complicated, as I wrote a Node script that maps the data into the new format and inserts it into Supabase.

I just can't figure out how to migrate the auth users.
Also, I'd like to change the id type for every row, both auth and db. I'd like to use UUID with Supabase.
And I'm aware I'm gonna have to do some mapping to keep the relationships intact.

Has any tried something similar or has any idea on how to do it?

Thanks!

**Comment by cdedreuille** [comment, marked as answer]
Hi @marcelscruz. Moving Auth users from Firebase is a bit complex as you can't export passwords. I ended up creating a flow on login that will create a new Supabase user when they log in. ([discussion here](https://github.com/supabase/supabase/discussions/1264)).

**Reply by marcelscruz** [reply]
Thanks for the reply, @cdedreuille. I'm moving to that thread, so we can continue the discussion.
